### ISSUE TITLE
unity-base/unity-7.4.0_p0_p01_p2016052601 build error

### ISSUE THREAD

**Issue by and3k** [shiznix/unity-gentoo#137]
Hi!

When I try upgrading unity, I get this error:

```
[ 13%] Building CXX object UnityCore/CMakeFiles/unity-core-6.0.dir/ScopeData.cpp.o
In file included from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:21:0:
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:45:17: error: ‘vector’ is not a member of ‘std’
   nux::Property<std::vector<std::string>> keywords;
                 ^
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:45:17: note: suggested alternatives:
In file included from /usr/include/boost/mpl/aux_/include_preprocessed.hpp:37:0,
                 from /usr/include/boost/mpl/vector.hpp:46,
                 from /usr/include/boost/math/policies/policy.hpp:14,
                 from /usr/include/boost/math/special_functions/math_fwd.hpp:28,
                 from /usr/include/boost/math/special_functions/sign.hpp:17,
                 from /usr/include/boost/lexical_cast/detail/inf_nan.hpp:34,
                 from /usr/include/boost/lexical_cast/detail/converter_lexical_streams.hpp:63,
                 from /usr/include/boost/lexical_cast/detail/converter_lexical.hpp:54,
                 from /usr/include/boost/lexical_cast/try_lexical_convert.hpp:42,
                 from /usr/include/boost/lexical_cast.hpp:32,
                 from /usr/include/Nux-4.0/NuxCore/PropertyTraits.h:31,
                 from /usr/include/Nux-4.0/NuxCore/Property.h:25,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:24,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:21:
/usr/include/boost/mpl/aux_/preprocessed/gcc/vector.hpp:22:8: note:   ‘boost::mpl::vector’
 struct vector;
        ^
In file included from /usr/include/boost/lexical_cast/detail/converter_lexical.hpp:52:0,
                 from /usr/include/boost/lexical_cast/try_lexical_convert.hpp:42,
                 from /usr/include/boost/lexical_cast.hpp:32,
                 from /usr/include/Nux-4.0/NuxCore/PropertyTraits.h:31,
                 from /usr/include/Nux-4.0/NuxCore/Property.h:25,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:24,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:21:
/usr/include/boost/container/container_fwd.hpp:108:7: note:   ‘boost::container::vector’
 class vector;
       ^
In file included from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:21:0:
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:45:17: error: ‘vector’ is not a member of ‘std’
   nux::Property<std::vector<std::string>> keywords;
                 ^
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:45:17: note: suggested alternatives:
In file included from /usr/include/boost/mpl/aux_/include_preprocessed.hpp:37:0,
                 from /usr/include/boost/mpl/vector.hpp:46,
                 from /usr/include/boost/math/policies/policy.hpp:14,
                 from /usr/include/boost/math/special_functions/math_fwd.hpp:28,
                 from /usr/include/boost/math/special_functions/sign.hpp:17,
                 from /usr/include/boost/lexical_cast/detail/inf_nan.hpp:34,
                 from /usr/include/boost/lexical_cast/detail/converter_lexical_streams.hpp:63,
                 from /usr/include/boost/lexical_cast/detail/converter_lexical.hpp:54,
                 from /usr/include/boost/lexical_cast/try_lexical_convert.hpp:42,
                 from /usr/include/boost/lexical_cast.hpp:32,
                 from /usr/include/Nux-4.0/NuxCore/PropertyTraits.h:31,
                 from /usr/include/Nux-4.0/NuxCore/Property.h:25,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:24,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:21:
/usr/include/boost/mpl/aux_/preprocessed/gcc/vector.hpp:22:8: note:   ‘boost::mpl::vector’
 struct vector;
        ^
In file included from /usr/include/boost/lexical_cast/detail/converter_lexical.hpp:52:0,
                 from /usr/include/boost/lexical_cast/try_lexical_convert.hpp:42,
                 from /usr/include/boost/lexical_cast.hpp:32,
                 from /usr/include/Nux-4.0/NuxCore/PropertyTraits.h:31,
                 from /usr/include/Nux-4.0/NuxCore/Property.h:25,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:24,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:21:
/usr/include/boost/container/container_fwd.hpp:108:7: note:   ‘boost::container::vector’
 class vector;
       ^
In file included from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:21:0:
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:45:34: error: template argument 1 is invalid
   nux::Property<std::vector<std::string>> keywords;
                                  ^
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:45:40: error: expected unqualified-id before ‘>’ token
   nux::Property<std::vector<std::string>> keywords;
                                        ^
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp: In static member function ‘static unity::dash::ScopeData::Ptr unity::dash::ScopeData::ReadProtocolDataForId(const string&, unity::glib::Error&)’:
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:74:5: error: ‘vector’ is not a member of ‘std’
     std::vector<std::string> keywords;
     ^
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:74:5: note: suggested alternatives:
In file included from /usr/include/boost/mpl/aux_/include_preprocessed.hpp:37:0,
                 from /usr/include/boost/mpl/vector.hpp:46,
                 from /usr/include/boost/math/policies/policy.hpp:14,
                 from /usr/include/boost/math/special_functions/math_fwd.hpp:28,
                 from /usr/include/boost/math/special_functions/sign.hpp:17,
                 from /usr/include/boost/lexical_cast/detail/inf_nan.hpp:34,
                 from /usr/include/boost/lexical_cast/detail/converter_lexical_streams.hpp:63,
                 from /usr/include/boost/lexical_cast/detail/converter_lexical.hpp:54,
                 from /usr/include/boost/lexical_cast/try_lexical_convert.hpp:42,
                 from /usr/include/boost/lexical_cast.hpp:32,
                 from /usr/include/Nux-4.0/NuxCore/PropertyTraits.h:31,
                 from /usr/include/Nux-4.0/NuxCore/Property.h:25,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:24,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:21:
/usr/include/boost/mpl/aux_/preprocessed/gcc/vector.hpp:22:8: note:   ‘boost::mpl::vector’
 struct vector;
        ^
In file included from /usr/include/boost/lexical_cast/detail/converter_lexical.hpp:52:0,
                 from /usr/include/boost/lexical_cast/try_lexical_convert.hpp:42,
                 from /usr/include/boost/lexical_cast.hpp:32,
                 from /usr/include/Nux-4.0/NuxCore/PropertyTraits.h:31,
                 from /usr/include/Nux-4.0/NuxCore/Property.h:25,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.h:24,
                 from /var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:21:
/usr/include/boost/container/container_fwd.hpp:108:7: note:   ‘boost::container::vector’
 class vector;
       ^
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:74:28: error: expected primary-expression before ‘>’ token
     std::vector<std::string> keywords;
                            ^
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:74:30: error: ‘keywords’ was not declared in this scope
     std::vector<std::string> keywords;
                              ^
/var/tmp/portage/unity-base/unity-7.4.0_p0_p01_p2016052601/work/UnityCore/ScopeData.cpp:86:11: error: ‘class unity::dash::ScopeData’ has no member named ‘keywords’
     data->keywords = keywords;
           ^
make[2]: *** [UnityCore/CMakeFiles/unity-core-6.0.dir/build.make:903: UnityCore/CMakeFiles/unity-core-6.0.dir/ScopeData.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3827: UnityCore/CMakeFiles/unity-core-6.0.dir/all] Error 2
make: *** [Makefile:128: all] Error 2
```

emerge --info:

```
Portage 2.3.0_rc1 (python 3.5.1-final-0, default/linux/amd64/13.0, gcc-5.4.0, glibc-2.23-r2, 3.18.2-gentoo-bela x86_64)
=================================================================
                         System Settings
=================================================================
System uname: Linux-3.18.2-gentoo-bela-x86_64-Intel-R-_Core-TM-_i7-4771_CPU_@_3.50GHz-with-gentoo-2.2
KiB Mem:    32884896 total,   3638356 free
KiB Swap:   35839996 total,  35839996 free
Timestamp of repository gentoo: Sun, 19 Jun 2016 16:00:01 +0000
sh bash 4.3_p42-r2
ld GNU ld (Gentoo 2.25.1 p1.1) 2.25.1
app-shells/bash:          4.3_p42-r2::gentoo
dev-java/java-config:     2.2.0-r3::gentoo
dev-lang/perl:            5.24.0-r1::gentoo
dev-lang/python:          2.7.11-r2::gentoo, 3.3.5-r8::gentoo, 3.4.4::gentoo, 3.5.1-r3::gentoo
dev-util/cmake:           3.5.2-r1::gentoo
dev-util/pkgconfig:       0.29.1::gentoo
sys-apps/baselayout:      2.2::gentoo
sys-apps/openrc:          0.21::gentoo
sys-apps/sandbox:         2.10-r2::gentoo
sys-devel/autoconf:       2.13::gentoo, 2.69-r2::gentoo
sys-devel/automake:       1.11.6-r2::gentoo, 1.12.6-r1::gentoo, 1.13.4-r1::gentoo, 1.14.1-r1::gentoo, 1.15-r2::gentoo
sys-devel/binutils:       2.25.1-r1::gentoo
sys-devel/gcc:            5.4.0::gentoo
sys-devel/gcc-config:     1.8-r1::gentoo
sys-devel/libtool:        2.4.6-r2::gentoo
sys-devel/make:           4.2.1::gentoo
sys-kernel/linux-headers: 4.6::gentoo (virtual/os-headers)
sys-libs/glibc:           2.23-r2::gentoo
Repositories:

gentoo
    location: /usr/portage
    sync-type: rsync
    sync-uri: rsync://rsync.gentoo.org/gentoo-portage
    priority: -1000

haskell
    location: /var/lib/layman/haskell
    sync-type: laymansync
    sync-uri: git://github.com/gentoo-haskell/gentoo-haskell.git
    masters: gentoo
    priority: 50

sabayon-tools
    location: /var/lib/layman/sabayon-tools
    sync-type: laymansync
    sync-uri: https://github.com/fusion809/sabayon-tools.git
    masters: gentoo
    priority: 50

steam-overlay
    location: /var/lib/layman/steam-overlay
    sync-type: laymansync
    sync-uri: git://github.com/anyc/steam-overlay.git
    masters: gentoo
    priority: 50

unity-gentoo
    location: /var/lib/layman/unity-gentoo
    sync-type: laymansync
    sync-uri: git://github.com/shiznix/unity-gentoo.git
    masters: gentoo
    priority: 50

and3k-sunrise
    location: /var/lib/overlays/and3k-sunrise
    masters: gentoo
    priority: 9999

ACCEPT_KEYWORDS="amd64 ~amd64"
ACCEPT_LICENSE="*"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-march=native -O2 -pipe"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/lib64/libreoffice/program/sofficerc /usr/share/config /usr/share/gnupg/qualified.txt"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo /etc/texmf/language.dat.d /etc/texmf/language.def.d /etc/texmf/updmap.d /etc/texmf/web2c"
CXXFLAGS="-march=native -O2 -pipe"
DISTDIR="/usr/portage/distfiles"
EMERGE_DEFAULT_OPTS="--jobs --load-average 8 --autounmask n"
FCFLAGS="-O2 -pipe"
FEATURES="assume-digests binpkg-logs config-protect-if-modified distlocks ebuild-locks fixlafiles merge-sync news parallel-fetch preserve-libs protect-owned sandbox sfperms strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-O2 -pipe"
GENTOO_MIRRORS="http://distfiles.gentoo.org"
LANG="en_GB.utf8"
LDFLAGS="-Wl,-O1 -Wl,--as-needed"
MAKEOPTS="-j8 -l8"
PKGDIR="/usr/portage/packages"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages --exclude=/.git"
PORTAGE_TMPDIR="/var/tmp"
USE="X a52 aac acl acpi alsa amd64 autoipd avahi ayatana bash-completion berkdb bidi blas blast bluetooth bookmarks branding bzip2 cairo cdda cjk clang cli colord conductor contrib cracklib crypt cups cxx dbus declarative dri dts dvd dvdr ebook eds emboss encode equalizer evo exif expat fam ffmpeg firefox flac fortran frei0r gcj gdbm gif glamor gnome gnome-keyring gnome-online-accounts gpg gpm gstreamer gtk hdf5 iconv icu inotify introspection ipv6 java jpeg kdenlive lapack lcms ldap libnotify libsecret mad matplotlib mdnsresponder-compat melt mmx mmxext mng modules mp3 mp4 mpeg multilib nautilus nbconvert ncurses networkmanager nls notebook nptl ntfs ogg opengl openmp pam pango parcheck pcre pdf png policykit postproc ppds printsupport pulseaudio python qt3support qt4 qt5 readline sdl seccomp semantic-desktop session spell sse sse2 sse3 sse4_1 ssl ssse3 staging startup-notification svg sync-plugin-portage systemd tcpd theora tiff tinfo truetype tty-helpers udev udisks unicode upnp upower usb utils v4l vaapi video vorbis vpx webkit widgets wifi wmf wxwidgets x264 xattr xcb xetex xkb xml xmp xv xvfb xvid zeitgeist zeroconf zlib" ABI_X86="64 32" ALSA_CARDS="ali5451 als4000 atiixp atiixp-modem bt87x ca0106 cmipci emu10k1x ens1370 ens1371 es1938 es1968 fm801 hda-intel intel8x0 intel8x0m maestro3 trident usb-audio via82xx via82xx-modem ymfpci" APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="kexi words flow plan sheets stage tables krita karbon braindump author" CAMERAS="ptp2" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="aes avx avx2 fma3 mmx mmxext popcnt sse sse2 sse3 sse4_1 sse4_2 ssse3" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf superstar2 timing tsip tripmate tnt ublox ubx" GRUB_PLATFORMS="efi-64" INPUT_DEVICES="keyboard mouse evdev" KERNEL="linux" LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" LINGUAS="en en_GB de de_AT" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php5-5" PYTHON_SINGLE_TARGET="python2_7" PYTHON_TARGETS="python2_7 python3_3 python3_4 python3_5" RUBY_TARGETS="ruby20 ruby21 ruby22 ruby23" USERLAND="GNU" VIDEO_CARDS="intel nvidia" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq steal rawnat logmark ipmark dhcpmac delude chaos account"
Unset:  CC, CPPFLAGS, CTARGET, CXX, INSTALL_MASK, LC_ALL, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, PORTAGE_RSYNC_EXTRA_OPTS, USE_PYTHON
```

Thanks!
Bela

**Comment by and3k** [comment]
Hm, after I unmasked some ebuilds because of unrelated things, and activated +pch globally, now it builds. Not sure what the cause is … Anyway, this can be closed as far as I'm concerned.

**Comment by shiznix** [comment]
Thanks Bela it's a valid bug, nux needs to be rebuilt when boost is upgraded.
Will commit a fix shortly that forces nux to rebuild on boost subslot change.

**Comment by shiznix** [comment]
Should be fixed in commit 6739bdf66ff8e9b0b6947732f7c1b8956519a7db
